core/class/jMQTTCmd.class.php

```diff
@@ -246,32 +246,34 @@ public function preSave() {
 
         }
 
-		// Reset auto_publish if info cmd
-		if ($this->getType() == 'info' && $this->getConfiguration('auto_publish', 0))
-			$this->setConfiguration('auto_publish', 0);
-		// Check "request" value if auto_publish enabled
-		if ($this->getType() == 'action' && $this->getConfiguration('auto_publish', 0)) {
+		// Reset autoPub if info cmd
+		if ($this->getType() == 'info' && $this->getConfiguration('autoPub', 0))
+			$this->setConfiguration('autoPub', 0);
+		// Check "request" value if autoPub enabled
+		if ($this->getType() == 'action' && $this->getConfiguration('autoPub', 0)) {
 			$req = $this->getConfiguration('request', '');
 			// Get all commands
 			preg_match_all("/#([0-9]*)#/", $req, $matches);
 			$cmds = array_unique($matches[1]);
 			// $value = '';
 			if (count($cmds) > 0) { // There are commands
+				$this_topic = $this->isJson() ? substr($this->getTopic(), 0, strpos($this->getTopic(), '{')) : $this->getTopic();
```

Comment:
"$this" represents an action cmd.
There shouldn't be some topic with {toto}.
Even if there is, we don't care

Reply:
Agreed !